exec/internal/compile/backend_amd64.go

```diff
@@ -365,6 +432,33 @@ func (b *AMD64Backend) emitBinaryI64(builder *asm.Builder, regs *dirtyRegs, op b
 	return nil
 }
 
+func (b *AMD64Backend) emitRHSConstOptimizedInstruction(builder *asm.Builder, regs *dirtyRegs, op byte, immediate uint64) error {
+
+	b.emitWasmStackLoad(builder, regs, x86.REG_AX)
+
+	prog := builder.NewProg()
+	prog.From.Type = obj.TYPE_CONST
+	prog.From.Offset = int64(immediate)
+	prog.To.Type = obj.TYPE_REG
+	prog.To.Reg = x86.REG_AX
+	switch op {
+	case ops.I64Add:
+		prog.As = x86.AADDQ
+	case ops.I64Sub:
+		prog.As = x86.ASUBQ
+	case ops.I64Shl:
+		prog.As = x86.ASHLQ
+	case ops.I64ShrU:
+		prog.As = x86.ASHRQ
+	default:
+		return fmt.Errorf("cannot handle op: %x", op)
```

Comment:
please prepend the package name.

Reply:
Nack: this is called from `AMD64Backend.Build()`, which will rewrite the error like so: ` fmt.Errorf("compile: amd64.emitRHSConstOptimizedInstruction: %v", err)`.

Reply:
fair enough.